mls-rs/src/group/snapshot.rs

```diff
@@ -159,24 +207,24 @@ where
     /// separately by calling [`Group::export_tree`].
     #[cfg_attr(not(mls_build_async), maybe_async::must_be_sync)]
     pub async fn write_to_storage_without_ratchet_tree(&mut self) -> Result<(), MlsError> {
-        let mut snapshot = self.snapshot();
+        let mut snapshot = self.snapshot()?;
         snapshot.state.public_tree.nodes = Default::default();
 
         self.state_repo.write_to_storage(snapshot).await
     }
 
-    pub(crate) fn snapshot(&self) -> Snapshot {
-        Snapshot {
+    pub(crate) fn snapshot(&self) -> Result<Snapshot, MlsError> {
```

Comment:
Why was it changed to return a `Result`? It seems `Ok` is always returned.

Reply:
Good catch, I will fix in a followup